### PR TITLE
fix(map_based_prediction): yaw rate can be overwritten by max function

### PR DESCRIPTION
**Pull request by danielsanchezaran**
## Description

small refactoring and fixing a bug with negative yaw rates. 

## Tests performed

Tested on Psim

## Effects on system behavior

Not applicable.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Review by YoshiRi — APPROVED** [review]
LGTM

File: perception/map_based_prediction/src/path_generator.cpp [hunk 1]
```diff
@@ -162,7 +162,7 @@ PredictedPath PathGenerator::generateStraightPath(const TrackedObject & object)
 {
   const auto & object_pose = object.kinematics.pose_with_covariance.pose;
   const auto & object_twist = object.kinematics.twist_with_covariance.twist;
-  const double ep = 0.001;
+  constexpr double ep = 0.001;
   const double duration = time_horizon_ + ep;
```
Comment (kyoichi-sugahara) [inline perception/map_based_prediction/src/path_generator.cpp:166]:
```suggestion
  constexpr double ep = 0.001;
  constexpr double duration = time_horizon_ + ep;
```

**Review by kyoichi-sugahara — APPROVED** [review]
LGTM!

File: perception/map_based_prediction/src/path_generator.cpp (hunk 1 above)
Comment (danielsanchezaran) [reply perception/map_based_prediction/src/path_generator.cpp:166]: @kyoichi-sugahara thanks for the suggestion, time_horizon_ cannot be decided at compile time so duration cannot be constexpr

Comment (kyoichi-sugahara) [reply perception/map_based_prediction/src/path_generator.cpp:166]: oh sorry

**Pull request merged**